### PR TITLE
Remove deprecated Elasticsearch passthrough query 

### LINKED ISSUE #13050: Remove deprecated query pass-through mechanism for Elasticsearch connector
To be removed: https://trino.io/docs/current/connector/elasticsearch.html?highlight=elasticsearch#pass-through-queries

The pass-through mechanism was deprecated in https://github.com/trinodb/trino/pull/12324 in favor of the `raw_query` table function.

### PR DESCRIPTION
**Pull request by kamil-rafalko**
## Description

This change removes deprecated Elasticsearch pass-through query from the Elasticearch connector

## Related issues, pull requests, and links
Fixes #13050

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Section
* Removed Elasticsearch pass-through query. ({issue}`#13050`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
Thx for your contribution @kamil-rafalko. Did you already sign CLA?

**Comment by kamil-rafalko** [comment]
@wendigo Yes I've signed and sent it to cla@trino.io as mentioned by cla-bot

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java [hunk 1]
```diff
@@ -167,39 +158,12 @@ public ElasticsearchTableHandle getTableHandle(ConnectorSession session, SchemaT
             String[] parts = tableName.getTableName().split(":", 2);
             String table = parts[0];
             Optional<String> query = Optional.empty();
-            ElasticsearchTableHandle.Type type = SCAN;
             if (parts.length == 2) {
-                // TODO this query pass-through mechanism is deprecated in favor of the `raw_query` table function.
-                //  it should be eventually removed: https://github.com/trinodb/trino/issues/13050
-                if (table.endsWith(PASSTHROUGH_QUERY_SUFFIX)) {
```
Comment (kasiafi) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java:174]: After we remove it, in a query like:
`SELECT * FROM es.default."<index>$query:<es-query>"`,
the whole chunk: `<index>$query` will be interpreted as the index name. Perhaps this is a valid index name. However, for the case when the index is not found, and someone was just trying to use the old pass-through mechanism, we might try to produce an error message with a hint.

**Review by kasiafi — COMMENTED** [review]
I'm concerned about removing this feature so soon. Probably, we should announce it on slack and wait so that users can migrate to `raw_query`.
@martint do you have any suggestions on timing?

**Comment by martint** [comment]
We shouldn’t remove it at least until there’s a way to perform access control checks for table functions using the built in access control mechanisms.

**Comment by kamil-rafalko** [comment]
@kasiafi Is there any way of informing users that they're using a deprecated feature? If we can do something like this, maybe it's worth thinking about adding such information together with suggestion on how to replace it before removing it at all.

**Comment by kasiafi** [comment]
@kamil-rafalko the reasonable way would be to produce a warning. However, we don't collect warnings from the Elasticsearch plugin, and we don't want to reason about the catalog or table name in the StatemetAnalyzer.

**Comment by kamil-rafalko** [comment]
@kasiafi I've committed your suggestions, but of course I leave decision to you of how to release this. If you decide to remove this feature by adding intermediate step which will be warning user in some way I can help with implementing it. :)

**Comment by kokosing** [comment]
@kamil-rafalko Welcome in the community! Please take a look at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md to make it easier to pass the code review process.

**Comment by kokosing** [comment]
How about we hide the functionality behind the feature toggle first (to stress that it is deprecated). So it would be disabled by default and user (admin) would need to accept that that they enable deprecated features. WDYT?

**Comment by kokosing** [comment]
We also need https://github.com/trinodb/trino/pull/13713 to control the access to table function as this table function is very powerful so it cannot be allowed to use to anywone.

It is a pity that this query passthrough was exposed without even a feature toggle so far.

**Comment by kasiafi** [comment]
@kokosing @martint is it time to proceed with this PR?

**Comment by kokosing** [comment]
@kamil-rafalko Would you like to rebase the PR so we can move forward with it?

**Comment by kamil-rafalko** [comment]
@kokosing Sure, will rebase it. Sorry for the late reply :)

**Comment by kamil-rafalko** [comment]
Rebased and removed pass-through query feature flag. Please take a look  :)

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java [hunk 2]
```diff
@@ -110,16 +105,13 @@
...
 import static java.util.Collections.emptyIterator;
 import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
 
 public class ElasticsearchMetadata
         implements ConnectorMetadata
 {
-    private static final ObjectMapper JSON_PARSER = new ObjectMapperProvider().get();
-
     private static final String PASSTHROUGH_QUERY_SUFFIX = "$query";
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java:115]: I would remove this all together. I guess the error message that table is not found is good enough.

**Review by kokosing — COMMENTED** [review]
I would squash commits. Revert after such long time since the introduction is not an revert, but a change IMO. Also the change at that time was ok as conditions were different, and revert suggest it was not.

**Comment by kamil-rafalko** [comment]
@kokosing Pushed changes according to your comments, please take a look again

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java [hunk 3]
```diff
@@ -171,44 +159,12 @@ public ElasticsearchTableHandle getTableHandle(ConnectorSession session, SchemaT
             String[] parts = tableName.getTableName().split(":", 2);
             String table = parts[0];
             Optional<String> query = Optional.empty();
-            ElasticsearchTableHandle.Type type = SCAN;
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java:174]: Please remove `ElasticsearchTableHandle.Type` too.

**Review by kokosing — APPROVED** [review]
% comment

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java (hunk 3 above)
Comment (kokosing) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java:174]: Removing this class will trigger more code removal

Comment (kokosing) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java:174]: Nevermind. It is actually used for table function.

**Comment by kokosing** [comment]
I will keep it open for few more days to give others a chance to take a look. Then in case nobody has issues with it I will merge it.

**Review by kasiafi — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Thanks!

**Pull request merged**